src/query/spec.ts

```diff
@@ -108,8 +111,9 @@ export function stack(specQ: SpecQuery): StackProperties & {fieldEncQ: EncodingQ
   const yEncQ = specQ.encodings.reduce((f, encQ: EncodingQuery) => {
     return f || (encQ.channel === Channel.Y ? encQ : null);
   }, null);
-  const xIsAggregate = !!xEncQ && (!!xEncQ.aggregate || !!xEncQ.autoCount);
-  const yIsAggregate = !!yEncQ && (!!yEncQ.aggregate || !!yEncQ.autoCount);
+  // TODO(akshatsh): check this
+  const xIsAggregate = isFieldQuery(xEncQ) && (!!xEncQ && (!!xEncQ.aggregate || !!xEncQ.autoCount));
```

Comment:
Since your `isFieldQuery` already checks if it's not null/undefined, we no longer need `!!xEncQ` clause.   (Same for `!!yEncQ`.)